### PR TITLE
`groupArray` returns cannot be nullable

### LINKED ISSUE #48462: Should `returns_default_when_only_null` be true for  `groupArray`
> Make sure to check documentation https://clickhouse.com/docs/en/ first. If the question is concise and probably has a short answer, asking it in [community Slack](https://join.slack.com/t/clickhousedb/shared_invite/zt-1gh9ds7f4-PgDhJAaF8ad5RbWBAAjzFg) is probably the fastest way to find the answer. For more complicated questions, consider asking them on StackOverflow with "clickhouse" tag https://stackoverflow.com/questions/tagged/clickhouse 

> If you still prefer GitHub issues, remove all this text and ask your question here.

Since `array` type could not be null, should the property `returns_default_when_only_null` for `groupArray` be true ?

```C++
void registerAggregateFunctionGroupArray(AggregateFunctionFactory & factory)
{
    AggregateFunctionProperties properties = { .returns_default_when_only_null = false, .is_order_dependent = true };

    factory.registerFunction("groupArray", { createAggregateFunctionGroupArray<false>, properties });
    factory.registerFunction("groupArraySample", { createAggregateFunctionGroupArraySample, properties });
    factory.registerFunction("groupArrayLast", { createAggregateFunctionGroupArray<true>, properties });
}

```

### PR DESCRIPTION
**Pull request by lgbo-ustc**
### Changelog category (leave one):
- Bug Fix

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixed `Nested type Array(T) cannot be inside Nullable type` for `groupArray` function.

closed #48462

### CODE REVIEW

File: tests/queries/0_stateless/02713_group_array_nullable.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+SELECT groupArray(number) FROM numbers(10) SETTINGS aggregate_functions_null_for_empty = 1;
+SELECT groupArrayLast(2)(number) FROM numbers(10) SETTINGS aggregate_functions_null_for_empty = 1;
```
Comment (nickitat) [inline tests/queries/0_stateless/02713_group_array_nullable.sql:2]:
```suggestion
SET aggregate_functions_null_for_empty = 1;

SELECT groupArray(1);
SELECT groupArray(number) FROM numbers(10);
SELECT groupArrayLast(2)(number) FROM numbers(10);
```

**Comment by lgbo-ustc** [comment]
change the result of `groupArray(NULL)` from `null` to `[]`

**Review by nickitat — APPROVED** [review]
let's pray we didn't break backward compatibility

**Comment by azat** [comment]
This does break the binary format (updated test references shows that there are problems), so I'm suggesting to revert it for now - #49971 (yes one release already affected, but I don't see a way to fix this in a binary compatible way now, and I hope that not a lot people affected by this), and fix it without breaking compatibility (also I found more changes of the on disk format for aggregate functions over Nullable, that comes from #11593)

Thoughts?

**Comment by lgbo-ustc** [comment]
> This does break the binary format (updated test references shows that there are problems), so I'm suggesting to revert it for now - #49971 (yes one release already affected, but I don't see a way to fix this in a binary compatible way now, and I hope that not a lot people affected by this), and fix it without breaking compatibility (also I found more changes of the on disk format for aggregate functions over Nullable, that comes from #11593)
> 
> Thoughts?

It's a compatibility issue that could happen on master branch with following query ?
```sql
select hex(groupArrayState(toNullable(1)))
```

**Pull request merged**